maven/lib/dependabot/maven/file_parser/maven_dependency_parser.rb

```diff
@@ -0,0 +1,135 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+
+require "dependabot/maven/file_parser"
+require "dependabot/maven/native_helpers"
+
+module Dependabot
+  module Maven
+    class FileParser
+      class MavenDependencyParser
+        extend T::Sig
+        require "dependabot/file_parsers/base/dependency_set"
+
+        DEPENDENCY_OUTPUT_FILE = "dependency-tree-output.json"
+
+        sig do
+          params(dependency_files: T::Array[Dependabot::DependencyFile])
+            .returns(Dependabot::FileParsers::Base::DependencySet)
+        end
+        def build_dependency_set(dependency_files)
+          dependency_set = Dependabot::FileParsers::Base::DependencySet.new
+
+          # Copy only pom.xml files to a temporary directory to
+          # output the dependency tree without building the project
+          SharedHelpers.in_a_temporary_directory do |temp_path|
+            # Create a directory structure that maintains relative relationships
+            project_directory = create_directory_structure(dependency_files, temp_path.to_s)
+
+            dependency_files.each do |pom|
+              pom_path = File.join(project_directory, pom.name)
+              pom_dir = File.dirname(pom_path)
+              FileUtils.mkdir_p(pom_dir)
+              File.write(pom_path, pom.content)
+            end
+
+            Dir.chdir(project_directory) do
+              NativeHelpers.run_mvn_dependency_tree_plugin(DEPENDENCY_OUTPUT_FILE)
+            end
+
+            # mvn CLI outputs dependency tree for each pom.xml file, collect them
+            # add into single dependency set
+            dependency_files.each do |pom|
+              pom_path = File.join(project_directory, pom.name)
```

Comment:
In Gradle I had to do the equivalent of `File.join(temp_path, pom.directory, pom.name)` in order to get the full directory structure. I THINK you're handling this via project_directory but just in case not I thought I'd point it out.

Reply:
I think in Maven updater subdirectories path are part of the `.name` property, at least that's what I've found in my tests